profiles/common/modules/features/nexteuropa_trackedchanges/nexteuropa_trackedchanges.module

```diff
@@ -0,0 +1,747 @@
+              }
+              if ($is_moderated) {
+                // Workbench moderation case, we redirect directly to the
+                // current revision view.
+                $entity_path .= '/current-revision';
+                $link = l($raw, $entity_path, $link_options);
+              }
+              else {
+                $link = l($raw, $entity_path, $link_options);
+              }
+              $lang_value['rendered']['#markup'] = $link;
+              $result->field_field_netc_info_languages[$key2] = $lang_value;
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+/**
+ * Callback for getting netc_info related entity info human name.
+ *
+ * It supports Entity type, Bundle and Moderation state on netc_info data.
+ *
+ * @see nexteuropa_trackedchanges_entity_property_info_alter()
+ * @see hook_entity_property_info()
+ */
+function nexteuropa_trackedchanges_get_human_name($data, array $options, $name, $type, $info) {
+  $human_name = $name;
+  switch ($name) {
+    case 'rel_entity_state_human_name':
+      $human_name = t('N/A');
+      if (module_exists('workbench_moderation')) {
+        $state = $data->rel_entity_state;
+        $human_name = workbench_moderation_state_label($state);
+      }
+      break;
+
+    case 'rel_entity_bundle_name':
+      $type_info = entity_get_info($data->rel_entity_type);
+      $bundle = $data->rel_entity_bundle;
+      $bundle_info = $type_info['bundles'][$bundle];
+      $human_name = (isset($bundle_info['label'])) ? $bundle_info['label'] : $bundle;
+      break;
+
+    case 'rel_entity_type_name':
+      $type_info = entity_get_info($data->rel_entity_type);
+      $human_name = $type_info['label'];
+      break;
+  }
+
+  return $human_name;
+}
+
+/**
+ * Callback for getting netc_info related entity URL.
+ *
+ * @see nexteuropa_trackedchanges_entity_property_info_alter()
+ * @see hook_entity_property_info()
+ */
+function nexteuropa_trackedchanges_get_rel_uri($data, array $options, $name, $type, $info) {
+  $rel_entities = entity_load($data->rel_entity_type, array($data->rel_entity_id));
+  $uri = '';
+  if ($rel_entities) {
+    $rel_entity = reset($rel_entities);
+    $uri = entity_uri($data->rel_entity_type, $rel_entity);
+  }
+
+  return $uri ? url($uri['path'], $uri['options'] + $options) : '';
+}
+
+/**
+ * Get the 'nexteuropa_trackedchanges_info' entities related to an entity.
+ *
+ * @param string $entity_type
+ *   The type of the related entity.
+ * @param object $entity
+ *   The related entity.
+ *
+ * @return bool|array
+ *   An array of netc_info objects indexed by their ids.
+ *   When no results are found, FALSE is returned.
+ *
+ * @throws EntityMalformedException
+ *   The exception is thrown if $entity is malformed or if $entity_type
+ *   is empty.
+ */
+function nexteuropa_trackedchanges_info_load_by_entity($entity_type, $entity) {
+  if (!$entity_type) {
+    throw new EntityMalformedException('No entity type are set for this entity, 
```

Comment:
I hate doing such comment but... as it seems to be the trends nowadays... here we go: Does that error messages needs to be surrounded by the t() function to be properly translated ?